lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -730,7 +734,8 @@ def upload_grades_csv(_xmodule_instance_args, _entry_id, course_id, _task_input,
                 header = [section['label'] for section in gradeset[u'section_breakdown']]
                 rows.append(
                     ["id", "email", "username", "grade"] + header + cohorts_header +
```

Comment:
I think this syntax is a little bit clearer than trying to use `rows.extend`, though it's still a little confusing (obviously). `rows` is a list of lists. The `append` is adding a new row to the outer list. The `+` is constructing the row to be added. They're very different operations and should be kept separate.